daft/iceberg/iceberg_scan.py

```diff
@@ -84,7 +85,13 @@ def __init__(self, iceberg_table: Table, storage_config: StorageConfig) -> None:
         super().__init__()
         self._table = iceberg_table
         self._storage_config = storage_config
-        arrow_schema = schema_to_pyarrow(iceberg_table.schema())
+        iceberg_schema = iceberg_table.schema()
+        arrow_schema = schema_to_pyarrow(iceberg_schema)
+        self._field_id_mapping = {
+            # NOTE: A little dangerous here to use a private API
```

Comment:
To avoid the private API here, it would be best to create a visitor here similar to the `_IndexById` that returns a `Dict[int, PyField]` directly :)

Reply:
Done :)